challenges/dateAdder/date-adder.js

```diff
@@ -0,0 +1,24 @@
+function add(date, diff) {
+  if(typeof date !== 'object') throw 'Exception: input must be a date object';
+
+  const millisecondsLookup = {
+    s: 1000,
+    m: 1000 * 60,
+    h: 1000 * 60 * 60,
+    d: 1000 * 60 * 60 * 24,
+    w: 1000 * 60 * 60 * 24 * 7,
+    M: 0,
+    Y: 1000 * 60 * 60 * 24 * 365,
```

Comment:
what about leap year?

Reply:
I didn't account for it initially, but just learned it is really easy to take care of it using setFullYear() and getFullYear(). The pull request has been modified.